core/amber/src/main/scala/edu/uci/ics/texera/web/resource/dashboard/user/dataset/DatasetResource.scala

```diff
@@ -1068,67 +1070,100 @@ class DatasetResource {
       @QueryParam("path") pathStr: String,
       @Auth user: SessionUser
   ): Response = {
-    val uid = user.getUid
+    val (dataset, dsVersion) = resolveAndValidatePath(pathStr, user)
+    createZipResponse(dataset, dsVersion)
+  }
+
+  /**
+    * Retrieves a ZIP file for the latest dataset version.
+    *
+    * @param did The dataset ID.
+    * @param user The session user.
+    * @return A Response containing the latest dataset version as a ZIP file.
+    */
+  @GET
+  @Path("/{did}/latest-version-zip")
+  def retrieveDatasetLatestVersionZip(
+      @PathParam("did") did: UInteger,
+      @Auth user: SessionUser
+  ): Response = {
+    val (dataset, latestVersion) = getLatestVersionInfo(did, user)
+    createZipResponse(dataset, latestVersion)
+  }
+
+  private def resolveAndValidatePath(
+      pathStr: String,
+      user: SessionUser
+  ): (Dataset, DatasetVersion) = {
     val decodedPathStr = URLDecoder.decode(pathStr, StandardCharsets.UTF_8.name())
     val (_, dataset, dsVersion, _) =
       resolvePath(Paths.get(decodedPathStr), shouldContainFile = false)
 
-    withTransaction(context) { ctx =>
-      val did = dataset.getDid
-      val dvid = dsVersion.getDvid
-
-      if (!userHasReadAccess(ctx, did, uid)) {
-        throw new ForbiddenException(ERR_USER_HAS_NO_ACCESS_TO_DATASET_MESSAGE)
-      }
-
-      val targetDatasetPath = PathUtils.getDatasetPath(did)
-      val datasetVersion = getDatasetVersionByID(ctx, dvid)
-      val versionHash = datasetVersion.getVersionHash
-
-      val fileNodes = GitVersionControlLocalFileStorage.retrieveRootFileNodesOfVersion(
-        targetDatasetPath,
-        versionHash
-      )
+    validateUserAccess(dataset.getDid, user.getUid)
+    (dataset, dsVersion)
+  }
 
-      val streamingOutput = new StreamingOutput() {
-        override def write(outputStream: OutputStream): Unit = {
-          val zipOutputStream = new ZipOutputStream(outputStream)
+  private def getLatestVersionInfo(did: UInteger, user: SessionUser): (Dataset, DatasetVersion) = {
```

Comment:
After merging the two, I believe there is no need to have these separate functions

Reply:
I'll just leave these separate functions for readability.